dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -390,7 +390,7 @@ public T deleteHostedVersion(@ApiParam(hidden = true) @Auth User user,
         // If the version that's about to be deleted is the default version, unset it
         if (entry.getActualDefaultVersion().getName().equals(version)) {
             Optional<U> max = entry.getWorkflowVersions().stream().filter(v -> !Objects.equals(v.getName(), version))
-                    .max(Comparator.comparingLong(ver -> ver.getDate().getTime()));
+                    .max(Comparator.comparingLong(ver -> ver.getDate() != null ? ver.getDate().getTime() : ver.getDbCreateDate().getTime()));
```

Comment:
devil's advocate? 
What's the difference? Why not just always use dbcreatedate?

Reply:
last built comes from Quay.io, and is when the container was last built.

Reply:
That might open more questions. 

i.e. a workflow author might intentionally author new versions of the workflow that point to older Docker images to avoid a regression bug.  There might also be multiple versions that point to the same image (same built) but have "properly" sorted db create dates. Is there any case in which sorting by the Docker image is superior?

Reply:
Darn just realized this is abstract hosted, so for workflows too.

Reply:
Both actually set lastModified, would that be a better fallback? @denis-yuen

Reply:
Would it be ok if this goes into another ticket, this is a big blocker for any new PR changes.

Reply:
This code is just saying, if they delete a default version for a hosted entry, set the new default date to be the most recently created. So I actually think we can just always use dbCreateDate. Does that make sense?